src/BuildScriptGenerator.Common/VersionInfo.cs

```diff
@@ -0,0 +1,41 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Oryx.BuildScriptGenerator.Common
+{
+    public class VersionInfo : IComparable<VersionInfo>
+    {
+        public string displayVersion { get; }
+        public SemVer.Version semanticVersion { get;  }
+
+        public VersionInfo(string displayVersion)
+        {
+            this.displayVersion = displayVersion;
+            this.semanticVersion = this.ToSemanticVersion(this.displayVersion);
+        }
+
+        private SemVer.Version ToSemanticVersion(string displayVersion)
+        {
+            string semanticVersionStr = displayVersion;
+
+            if (!displayVersion.Contains('-') && !Regex.IsMatch(displayVersion, @"^\d+\.\d+\.\d+$"))
```

Comment:
Avoid using regular expressions. Prefer parsing the code

Reply:
Why do we prefer to parse the code instead of using regex, I actually not understand the reason, I think it looks clean and covered all the cases.. but will work on try parsing it